File: app/src/main/java/sk/styk/martin/apkanalyzer/util/JsonSerializationUtils.java
package sk.styk.martin.apkanalyzer.util;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import java.text.DateFormat;

import static com.google.gson.FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES;

/**
 * @author Martin Styk
 * @version 06.11.2017.
 */
public class JsonSerializationUtils {

    private Gson gson;

    public JsonSerializationUtils() {
        this.gson = new GsonBuilder()
                .setFieldNamingPolicy(LOWER_CASE_WITH_UNDERSCORES)
                .setDateFormat("yyyy-MM-dd HH:mm a z").create();
    }

    public String serialize(Object object) {
        return gson.toJson(object);
    }
}

File: app/src/main/java/sk/styk/martin/apkanalyzer/util/JsonSerializationUtils.kt
package sk.styk.martin.apkanalyzer.util

import com.google.gson.FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES
import com.google.gson.GsonBuilder

/**
 * @author Martin Styk
 * @version 06.11.2017.
 */
class JsonSerializationUtils {

    val gson = GsonBuilder()
            .setFieldNamingPolicy(LOWER_CASE_WITH_UNDERSCORES)
            .setDateFormat("yyyy-MM-dd HH:mm a z").create()

    fun serialize(`object`: Any): String {
        return gson.toJson(`object`)
    }

    inline fun <reified T> deserialize(json: String): T {
        return gson.fromJson(json, T::class.java)
    }
}
